streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindows.java

```diff
@@ -214,7 +216,10 @@ public long gracePeriodMs() {
         // NOTE: in the future, when we remove maintainMs,
         // we should default the grace period to 24h to maintain the default behavior,
         // or we can default to (24h - size) if you want to be super accurate.
-        return graceMs != -1 ? graceMs : maintainMs() - size();
+        if (graceMs != EMPTY_GRACE_PERIOD) {
+            return graceMs;
+        }
+        return maintainDurationMs > sizeMs ? maintainDurationMs - sizeMs : 0;
```

Comment:
nit: how about
```suggestion
        return Math.max(maintainDurationMs - sizeMs, 0);
```
for a slight improvement in readability?